### ISSUE TITLE
(Debian) owncloud-client 2.3.3-8388 Segfault (just before suspend mode)

### ISSUE THREAD

**Issue by iceroot82** [owncloud/client#6072]
### Expected behaviour
Putting my notebook into suspend, after the notebook is waking up again, the owncloud-client should be still there

### Actual behaviour
When i am putting my Notebook into suspend mode, owncloud-clinet ist not active anymore after the notebook was coming back from sleep. In /var/log/syslog i found a segfault

Oct  2 18:01:43 bestian kernel: [160716.677254] owncloud[22613]: segfault at 7ffef5cdafc0 ip 00007faf25e14c8b sp 00007ffef5cdafb0 error 6 in libc-2.24.so[7faf25d9c000+195000]

right before the network was already put off (if i read these lines correctly)

Oct  2 18:01:42 bestian NetworkManager[665]: <info>  [1506960102.7111] dhcp6 (eth0): state changed unknown -> bound
Oct  2 18:01:42 bestian nm-dispatcher: req:11 'dhcp6-change' [eth0]: new request (1 scripts)
Oct  2 18:01:42 bestian nm-dispatcher: req:11 'dhcp6-change' [eth0]: start running ordered scripts...
Oct  2 18:01:42 bestian NetworkManager[665]: <info>  [1506960102.7179] dhcp6 (eth0): client pid 25342 exited with status 0
Oct  2 18:01:42 bestian NetworkManager[665]: <info>  [1506960102.7179] dhcp6 (eth0): state changed bound -> done

### Steps to reproduce
1. 
2. 
3. 

### Client configuration
Client version:
2.3.3-8338 

Operating system:
Debian GNU/Linux 9

OS language:
German

Qt version used by client package (Linux only, see also Settings dialog):
5.6.2
Gebaut von der GIT-Revision 57bc79 auf Sep 5 2017, 18:36:45 verwendet Qt 5.6.2, OpenSSL 1.0.2l 25 May 2017

Client package (From ownCloud or distro) (Linux only):
owncloud-client:
  Installiert:           2.3.3-8338
  Installationskandidat: 2.3.3-8338
  Versionstabelle:
 *** 2.3.3-8338 500
        500 http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/Debian_9.0  Packages

Installation path of client:
/usr/bin/owncloud

### Logs

Oct  2 18:01:42 bestian NetworkManager[665]: <info>  [1506960102.7111] dhcp6 (eth0): state changed unknown -> bound
Oct  2 18:01:42 bestian nm-dispatcher: req:11 'dhcp6-change' [eth0]: new request (1 scripts)
Oct  2 18:01:42 bestian nm-dispatcher: req:11 'dhcp6-change' [eth0]: start running ordered scripts...
Oct  2 18:01:42 bestian NetworkManager[665]: <info>  [1506960102.7179] dhcp6 (eth0): client pid 25342 exited with status 0
Oct  2 18:01:42 bestian NetworkManager[665]: <info>  [1506960102.7179] dhcp6 (eth0): state changed bound -> done
Oct  2 18:01:43 bestian kernel: [160716.677254] owncloud[22613]: segfault at 7ffef5cdafc0 ip 00007faf25e14c8b sp 00007ffef5cdafb0 error 6 in libc-2.24.so[7faf25d9c000+195000]

**Comment by guruz** [comment]
@iceroot82 Could you try to give us a more complete backtrace?

You could run the client from a console/terminal/shell with ```gdb owncloud```. Then try to reproduce the issue. Then if it crashes, you should be thrown into the gdb input. There you can type: ```thread apply all bt``` and give us the output. Thanks.

@dschmidt @jnweiger (Not sure if this is the issue but could be) Are we shipping the Qt bearer plugins in **our** Linux Qt package? Because I think we should not, we explicitely don't ship them on macOS and Windows.

**Comment by iceroot82** [comment]
@guruz 
of course and thank you four your response. I could easily reproduce the issue and hopefully i was using gdb correctly

https://pastebin.com/Pmssjv8p

At Line 52 it is the output i got directly after the wake up (Datei oder Verzeichnis nicht gefunden means file not found).

if you need further information please give me a sign.

Greetings from germany

**Comment by guruz** [comment]
(btw github also has a pastebin on http://gist.github.com :-)

Are you sure you don't have the backtrace of thread 1?

We should see the ```#0 #1 #2 #3``` etc similar to other threads?

```
Thread 1 "owncloud" received signal SIGSEGV, Segmentation fault.
0x00007ffff273ac8b in _int_malloc (av=av@entry=0x7ffff2a5bb00 <main_arena>, bytes=bytes@entry=25) at malloc.c:3387
3387    malloc.c: Datei oder Verzeichnis nicht gefunden.   (Means file not found)
```

**Comment by guruz** [comment]
@iceroot82 Just wondering, do you have a directory called ```bearer``` somewhere in ```/opt/ownCloud/qt-5.6.2/``` ? What happens if you move that directory to ```bearer.old``` and retry this? (also restarting the client)

**Comment by iceroot82** [comment]
@guruz i should have pasted all lines i got. Next time i will use the one from github :)

I had the directory you asked for
michael@bestian:/opt/ownCloud/qt-5.6.2$ find . -name bearer
./lib/x86_64-linux-gnu/qt5/plugins/bearer

I renamed it as you suggested, restarted the client and then did my suspend-testing. **20 times it was successful, so it seems to be fixed**

I did not find any packages containing that directory (inside the owncloud dir, just at another place)

michael@bestian:/opt/ownCloud/qt-5.6.2$ apt-file search qt5/plugins/bearer

libqt5network5: /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqconnmanbearer.so
libqt5network5: /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqgenericbearer.so
libqt5network5: /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqnmbearer.so

libqt5network5 is installed on my system btw

For me the issue seems to be fixed now, can i support you with additional information? or did you got everything you need on your site?

And thank you very much for your support, very helpful and very clear instructions :)

**Comment by jnweiger** [comment]
@guruz If I understand correctly, `configure -no-feature-bearermanagement ...` should do the trick. Right?

**Comment by guruz** [comment]
@jnweiger Not necessarily. :-( The configure would also add some ```#ifdef``` into the Qt code that makes it do different code paths.
You'd need to check how we compile Qt for Windows and macOS. It might be that we don't configure anything there (so compile with bearer management) and then just not ship the plugins (I'm sure about the latter part)

**Comment by jnweiger** [comment]
Disabling at compile time with --no-feature-bearermanagement does not work.
This is intrusive to the code. It causes e.g.

 204s] /home/abuild/rpmbuild/BUILD/owncloudclient-2.4.0/src/gui/application.h:120:5: error: 'QNetworkConfigurationManager' does not name a type

errors. We keep bearer enabled, but wipe the plugins directory. That should be safe as our Qt562 only collects plugin from its own plugin directory within /opt, and not from the system provided qt, if any.

**Comment by jnweiger** [comment]
Disabling at compile time with --no-feature-bearermanagement does not work.
This is intrusive to the code. It causes e.g.

 204s] /home/abuild/rpmbuild/BUILD/owncloudclient-2.4.0/src/gui/application.h:120:5: error: 'QNetworkConfigurationManager' does not name a type

errors. We keep bearer enabled, but wipe the plugins directory. That should be safe as our Qt562 only collects plugin from its own plugin directory within /opt, and not from the system provided qt, if any.

**Comment by jnweiger** [comment]
Current 2.4.0~alpha1 repo https://build.opensuse.org/package/show/isv:ownCloud:community:testing has 
ocqt562-libqt5network5:amd64 (5.6.2-1+22.1) which is a lower version than the same package from isv:ownCloud:desktop (5.6.2-13.1) according debian standards. 

Bumping the build release number to 5.6.2-14+XX.Y so that both repos
* https://build.opensuse.org/package/show/isv:ownCloud:desktop and
* https://build.opensuse.org/package/show/isv:ownCloud:community:testing 

can be registered and the latter latter takes precedence also with regard to the dependencies.

**Comment by jnweiger** [comment]
@iceroot82 can you re-test with https://software.opensuse.org//download.html?project=isv:ownCloud:community:testing&package=owncloud-client
this should be better... :-)

**Comment by iceroot82** [comment]
@jnweiger sure. I installed the Package from the Repo you named

michael@bestian:/opt/ownCloud$ apt-cache policy owncloud-client
owncloud-client:
  Installiert:           2.4.0~alpha1-8472
  Installationskandidat: 2.4.0~alpha1-8472
  Versionstabelle:
 *** 2.4.0~alpha1-8472 500
        500 http://download.opensuse.org/repositories/isv:/ownCloud:/community:/testing/Debian_9.0  Packages
        100 /var/lib/dpkg/status
     2.2.4+dfsg-2 500
        500 http://httpredir.debian.org/debian stretch/main amd64 Packages

The bearer directory was not created, only my old renamed one is there

michael@bestian:/opt/ownCloud$ find . -name bearer*
./qt-5.6.2/lib/x86_64-linux-gnu/qt5/plugins/bearer.old

I restarted the client and then testes the suspend/wake-up 10 times. Everything works fine, so from my point of view the new package is working great :)

Thank you all for your support and fixing the issue

**Comment by jnweiger** [comment]
@iceroot82 can you re-test with https://software.opensuse.org//download.html?project=isv:ownCloud:community:testing&package=owncloud-client
this should be better... :-)

**Comment by iceroot82** [comment]
@jnweiger thats the package i used for my last test or did i get something wrong?

`root@bestian:/home/michael# apt-cache policy owncloud-client
owncloud-client:
  Installiert:           2.4.0~alpha1-8472
  Installationskandidat: 2.4.0~alpha1-8472
  Versionstabelle:
 *** 2.4.0~alpha1-8472 500
        500 http://download.opensuse.org/repositories/isv:/ownCloud:/community:/testing/Debian_9.0  Packages
        100 /var/lib/dpkg/status
`

**Comment by jnweiger** [comment]
@iceroot82 ah, sorry I forgot to tell you the important details:
The bearer plugins do not come from package owncloud-client, but from package ocqt562-libqt5network5!
So the trick is to update ocqt562-libqt5network5 -- (But I am not exactly sure how your manual move is handled then :-) )

**Comment by iceroot82** [comment]
@jnweiger thank you for the information. I just used apt-get install owncloud-client to update that package but that was not updating the dependencies. (my fault)
Now with dist-upgrade i updated the package you name as well

The following packages will be updated:
  ocqt562-libqt5printsupport5 ocqt562-libqt5sql5 ocqt562-libqt5sql5-sqlite ocqt562-libqt5webkit5 ocqt562-libqt5widgets5 ocqt562-libqt5xml5
  owncloud-client owncloud-client-caja owncloud-client-l10n owncloud-client-overlays-icons

No new bearer Stuff inside

michael@bestian:/opt/ownCloud$ find . -name bearer*
./qt-5.6.2/lib/x86_64-linux-gnu/qt5/plugins/bearer.old

I also removed owncloud-client, then i used apt-get autoremove to have a clean base. I deleted 

sudo rm -r /opt/ownCloud/qt-5.6.2/lib/x86_64-linux-gnu/qt5/plugins/bearer.old

Then i installed owncloud-client again, which was pulling ocqt562-libqt5network5, noth from the testing repo

And there is no bearer stuff :)

michael@bestian:/opt/ownCloud$ find . -name bearer*
michael@bestian:/opt/ownCloud$ 

Do you need any other tests or information?

**Comment by jnweiger** [comment]
That confirms my fix. Thanks!
Let's close here. Reopen if you can produce bearer related crashes again :-)

@guruz fixed in 2.4.0 -- please reopen, if this is needed in 2.3.4 too.
